CorsixTH/Lua/dialogs/fullscreen/progress_report.lua

```diff
@@ -49,52 +48,35 @@ function UIProgressReport:UIProgressReport(ui)
 
   -- Selected hospital number
   self.selected = 1
+  local hospital = world.hospitals[self.selected]
 
+  -- Collect which criteria to show here, draw columns in draw
   -- Add the icons for the criteria
   local x = 263
-  local world_goals = world.goals
-  for _, tab in ipairs(world_goals) do
-    local crit_name = world.level_criteria[tab.criterion].name
-    local res_value = world_goals[crit_name].win_value
-    world_goals[crit_name].visible = true
-    -- Special case for money, subtract loans
-    local cur_value = hospital[crit_name]
-    if crit_name == "balance" then
-      cur_value = cur_value - hospital.loan
-    end
-    if world_goals[crit_name].lose_value then
-      world_goals[crit_name].red = false
-
-      if cur_value < world_goals[crit_name].boundary then
-        world_goals[crit_name].red = true
-        res_value = world_goals[crit_name].lose_value
-        -- TODO: Make the ugly workaround for the special case "percentage_killed" better
-        if crit_name:find("killed") then
-          res_value = nil
-          world_goals[crit_name].visible = false
-        end
-      elseif not world_goals[crit_name].win_value then
-        world_goals[crit_name].visible = false
+  if not world.free_build_mode then
```

Comment:
Can't you have empty win and lose goals instead?

That avoids having this check here.